src/config.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    // we assume the value was set on the CLI if the default value is
+    // not the same as that on the argv object as there is a very rare chance
+    // this happening
+    const wasValueSetOnCLI = typeof(argv[option]) !== 'undefined' &&
+      (argv[option] !== defaultValues[option]);
+    if (wasValueSetOnCLI) {
+      log.debug(`Favoring CLI: ${option}=${argv[option]} over ` +
+        `configuration: ${option}=${configObject[option]}`);
+      continue;
+    }
+    if (!argv.hasOwnProperty(option)) {
+      log.debug(`Ignoring configuration: ${option}=${configObject[option]} ` +
+        'because this is an unknown option');
+      continue;
+    }
+    newArgv[option] = configObject[option];
+  }
+  return newArgv;
+}
+
+export function loadJSConfigFile(filePath: string): Object {
+  log.debug(`Loading JS config file: ${filePath}`);
+  let configObject;
+  try {
+    configObject = requireUncached(filePath);
```

Comment:
I think this got lost in the churn. I had requested this change:
````js
const resolvedFilePath = path.resolve(filePath);
log.debug(
  `Loading JS config file: "${filePath}" (resolved to "${resolvedFilePath}")`);
let configObject;
try {
  configObject = requireUncached(resolvedFilePath);
}
...
````
Be sure to change all `filePath` variables to `resolvedFilePath`.

Without this change, a relative path might get mistaken for a node module. It's a little too hard to test this scenario so don't worry about adding a test.